posts/zzz_DO_NOT_EDIT_how__i__reb.../how__i__rebuilt_a__lost__ec_g__data__script_in__r.qmd

```diff
@@ -0,0 +1,172 @@
+
+```{r, eval = T, message = F, echo = F}         
+library(dplyr)
+library(pharmaversesdtm)
+
+eg %>%
+  select(USUBJID, EGTEST, VISIT, EGDTC, EGTPT, EGSTRESC) %>%
+  filter(USUBJID == "01-701-1015" & VISIT == "WEEK 2")
+```
+
+In this example, the `EGTPT` column indicates the timing of the test, while `EGTEST` specifies the type of test conducted. The numeric results are recorded in `EGSTRESN`, with the corresponding categorical values in `EGSTRESC`.
+
+### 2. Wrote the New R Script
+
+Armed with insights from my analysis, I set about writing a new R script to replicate the lost one. This involved a lot of trial and error, as I kept refining the code to ensure it generated a dataset that closely resembled the original ECG data in both structure and content. In order to give you, my reader, better understanding of the solution, I'll walk you through the main parts of the script.
+
+#### Loading Libraries and Data
+
+To begin, I loaded the necessary libraries and read in the vital signs (`vs`) dataset, which was essential for providing key clinical information about the participants. This data complemented the ECG measurements and allowed for a more comprehensive analysis of each subject's health status during the study. By setting a seed for the random data generation, I ensured that the process was reproducible, allowing others to verify my results and maintain consistency in future analyses. Additionally, the metatools package was loaded to facilitate adding labels to the variables later, which enhanced the readability of the dataset.
+
+```{r, eval = T, message = F}
+library(dplyr)
+library(metatools)
+library(pharmaversesdtm)
+
+data("vs")
+set.seed(123)
+```
+
+#### Extracting Unique Date/Time of Measurements
+
+Next, I extracted the unique combination of subject IDs, visit names, and visit dates from the `vs` dataset.
+
+```{r, eval = T}
+egdtc <- vs %>%
+  select(USUBJID, VISIT, VSDTC) %>%
+  distinct() %>%
+  rename(EGDTC = VSDTC)
+
+head(egdtc)
+```
+
+This data was used later to match the generated ECG data to the correct visit and time points.
+
+#### Generating a Grid of Patient Data
+
+Here, I created a grid of all possible combinations of subject IDs, test codes (e.g., `QT`, `HR`, `RR`, `ECGINT`), time points (e.g., after lying down, after standing), and visits. These combinations represented different test results collected across multiple visits.
+
+```{r, eval = T}
+eg <- expand.grid(
+  USUBJID = unique(vs$USUBJID),
+  EGTESTCD = c("QT", "HR", "RR", "ECGINT"),
+  EGTPT = c(
+    "AFTER LYING DOWN FOR 5 MINUTES",
+    "AFTER STANDING FOR 1 MINUTE",
+    "AFTER STANDING FOR 3 MINUTES"
+  ),
+  VISIT = c(
+    "SCREENING 1",
+    "SCREENING 2",
+    "BASELINE",
+    "AMBUL ECG PLACEMENT",
+    "WEEK 2",
+    "WEEK 4",
+    "AMBUL ECG REMOVAL",
+    "WEEK 6",
+    "WEEK 8",
+    "WEEK 12",
+    "WEEK 16",
+    "WEEK 20",
+    "WEEK 24",
+    "WEEK 26",
+    "RETRIEVAL"
+  ), stringsAsFactors = FALSE
+)
+
+head(eg)
```

Comment:
displaying `head(eg)` doesn't really show a useful snippet here - please could you revisit this?

![image](https://github.com/user-attachments/assets/5c2cbbae-7a4b-4005-990c-ea69c868566a)